app/javascript/packs/billboard.js

```diff
@@ -2,20 +2,25 @@ import { setupDisplayAdDropdown } from '../utilities/displayAdDropdown';
 
 // the term billboard can be synonymously interchanged with displayAd
 async function getBillboard() {
-  const placeholderElement = document.getElementsByClassName(
-    'js-display-ad-comments-container',
-  )[0];
+  const placeholderElements = document.getElementsByClassName(
+    'js-display-ad-container',
+  );
 
-  const { asyncUrl } = placeholderElement.dataset || {};
+  const promises = [...placeholderElements].map(generateDisplayAd);
+  await Promise.all(promises);
+}
+
+async function generateDisplayAd(element) {
+  const { asyncUrl } = element.dataset;
 
-  if (placeholderElement.innerHTML.trim() === '') {
+  if (element.innerHTML.trim() === '' && asyncUrl) {
```

Comment:
Just thinking here, but this potentially prevents us from adding placeholder content in the ad container later. What about setting a `data` attribute on the container after loading?

Reply:
This is a good point, thanks Joshua! I updated the code in a different way to allow placeholder content to be added but then clearing it out before we append a the generated display ad.

https://www.loom.com/share/85a37c14247446c582fd687a174efa1e